### ISSUE TITLE
Liquibase DDL modifications are not being marked as complete

### ISSUE THREAD

**Issue by salsakran** [metabase/metabase#3228]
From a bug report by @AndrewMagliozzi in #2883

It looks like a migration ran, but was not marked as completed in Liquibase. Prior to this, the app crashed after an OOM error.

```
2016-08-17T18:35:17.036935+00:00 heroku[web.1]: State changed from starting to up
2016-08-17T18:35:22.022575+00:00 app[web.1]: 08-17 18:35:22 �[1mDEBUG metabase.driver�[0m :: Registered driver �[34m:bigquery�[0m ??
2016-08-17T18:35:22.076591+00:00 app[web.1]: 08-17 18:35:22 �[1mDEBUG metabase.driver�[0m :: Registered driver �[34m:crate�[0m ??
2016-08-17T18:35:22.277335+00:00 app[web.1]: 08-17 18:35:22 �[1mDEBUG metabase.driver�[0m :: Registered driver �[34m:h2�[0m ??
2016-08-17T18:35:22.778680+00:00 app[web.1]: 08-17 18:35:22 �[1mDEBUG metabase.driver�[0m :: Registered driver �[34m:mongo�[0m ??
2016-08-17T18:35:22.798939+00:00 app[web.1]: 08-17 18:35:22 �[1mDEBUG metabase.driver�[0m :: Registered driver �[34m:mysql�[0m ??
2016-08-17T18:35:24.624514+00:00 app[web.1]: 08-17 18:35:24 �[1mINFO metabase.db�[0m :: �[36mVerifying postgres Database Connection ...�[0m
2016-08-17T18:35:27.268597+00:00 app[web.1]: 08-17 18:35:27 �[1mINFO metabase.db�[0m :: Verify Database Connection ... ✅
2016-08-17T18:36:27.378438+00:00 app[web.1]: 08-17 18:36:27 �[1mERROR liquibase�[0m :: migrations/liquibase.json: migrations/038_getting_started_guide.yaml::38::camsaul: Change Set migrations/038_getting_started_guide.yaml::38::camsaul failed.  Error: ERROR: column "points_of_interest" of relation "metabase_database" already exists [Failed SQL: ALTER TABLE public.metabase_database ADD points_of_interest TEXT]
2016-08-17T18:36:27.415639+00:00 app[web.1]: liquibase.exception.MigrationFailedException: Migration failed for change set migrations/038_getting_started_guide.yaml::38::camsaul:
2016-08-17T18:36:27.421378+00:00 app[web.1]:    at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
2016-08-17T18:36:27.422812+00:00 app[web.1]:    at metabase.core$_main.doInvoke(core.clj:255)
2016-08-17T18:36:27.424669+00:00 app[web.1]:    at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:107)
2016-08-17T18:36:27.428271+00:00 app[web.1]:    at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1273)
2016-08-17T18:36:28.097326+00:00 heroku[web.1]: Process exited with status 1
2016-08-17T18:36:28.114254+00:00 heroku[web.1]: State changed from up to crashed
```

**Comment by camsaul** [comment]
It may have been that the DB connection was set to auto commit, so when the server crashed it finalized the transaction even though all the things that were supposed to go in it weren't there yet

**Comment by salsakran** [comment]
If that's the case, it might be the reason locks are being left lying around.

**Comment by AndrewMagliozzi** [comment]
Any idea of what I can do to fix it and get the app back up?  I'm pretty technical and can manage most anything you throw my way.

Sorry for the slow reply, I had a nasty flu last week and fell behind on email.

**Comment by salsakran** [comment]
If you connect to Metabase's application database, you can try marking that migration as successfully completed and seeing if that works.

I'd strongly suggest backing up the database with a pg_dump before doing so.

**Comment by salsakran** [comment]
you should add an entry to the `databasechangelog`. The MD5 checksum for migraiton 38 is `7:5e32fa14a0c34b99027e25901b7e3255 `

alternatively, you can delete the new tables + columns in https://github.com/metabase/metabase/blob/master/resources/migrations/038_getting_started_guide.yaml and re-start.

**Comment by camsaul** [comment]
@AndrewMagliozzi 

The root cause of this issue is fixed by #3295; once Metabase 0.20 ships in the next few weeks you'll be able to fix your database automatically using the new `migrate force` command:

```bash
java -jar metabase.jar migrate force
```
